Assets/Scripts/GameplayObjects/IPooledObject.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public enum PooledObjectType
+{
+    Road,
+    Obstacle
+}
+
+public interface IPooledObject
```

Comment:
PooledObject means an object that an get the gameObject, in the end. it is too generic, any unity object can have a gameObject. 
I think you don't need this interface. correct me if I'm wrong: for example, in AsteroidFactory, when instantiating the object, you use getComponent in order to get the "underlying" object of the GameObject that is the asteroid (BaseObstacle) and return it. but you never use the BaseObstacle itself (it's properties or whatnot), you just end up using GetGameObject (I'm referring to GameplayObjectPool).